src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -5475,6 +5490,9 @@ int wm_vuldet_collect_agents_to_scan(wm_vuldet_t *vuldet) {
         w_strdup(agti_os_name, agents->os_name);
         if (agent_dist == FEED_WIN) {
             w_strdup(osinfo_os_release, agents->os_release);
+            w_strdup(osinfo_os_display_version, agents->os_display_version);
+            // The windows agent requires the architecture in the NVD format
+            wm_vuldet_normalize_architecture_NVD(agents->arch);
```

Comment:
With the suggestion made for `wm_vuldet_normalize_architecture_NVD` you will have to receive the return value in a `char*` string and after making use of it, de-allocate the memory.

Reply:
Fixed!